db/memtable.cc

```diff
@@ -720,9 +721,9 @@ static bool SaveValue(void* arg, const char* entry) {
 
 bool MemTable::Get(const LookupKey& key, std::string* value, Status* s,
                    MergeContext* merge_context,
-                   RangeDelAggregator* range_del_agg, SequenceNumber* seq,
-                   const ReadOptions& read_opts, ReadCallback* callback,
-                   bool* is_blob_index) {
+                   SequenceNumber* max_covering_tombstone_seq,
```

Comment:
If `max_covering_tombstone_seq` is nonzero can we return immediately, knowing that a covering range tombstone in any earlier memtable must be covering the key?

I noticed in the next PR you do this for SST files (well, using the file's largest seqnum), but I didn't see any early return for memtables.

Reply:
The main reason I didn't add this check to memtable is because I thought that having several memtables was fairly uncommon, and they're usually small enough that it doesn't matter whether or not a tombstone has been found (though I also didn't realize that `max_covering_tombstone_seq` being nonzero was sufficient; I think you're right that it should be, since the memtable, each L0 file, and each L1+ level cover disjoint bands of sequence numbers). I'll add this check to the second PR.